### PR TITLE
Add masking and behavior tests for some COP0 registers

### PR DESCRIPTION
**Pull request by bigbass1997**
This PR implements masking and behavior tests for the following COP0 registers:
* Random
* Wired
* Status
* Config
* Processor Revision Id
* Unused (7, 21, 22, 23, 24, 25, and 31)
* Parity Error
* Cache Error

I've included rustdocs for each test to explain any finer details of what behavior is being tested. Below are some extra notes regarding the design of this PR.

### Random
While it is possible to test only the masking by naively checking if Random is some value less than 64, this could produce a false-positive case. An emulator could allow writes to the lower 6 bits, but this is incorrect as the entire register is read-only. You also can't perform a simple "read, write, read, compare equality" test because the value is supposed to change on every instruction.

I'm planning to try to address this later in a different PR, to have different levels of scrutiny (not every emulator cares about being extremely accurate).

_Also note, despite the name, Random is definitely **not** random in any way._

### Status and Config
These regs, especially Config, have some constant 0 and 1 bits. So the test writes the existing value combined with whatever the opposite of these constant bits are. Then checks if the readback has these constant bits. Just writing all 1's or all 0's (like other masking tests) could produce false-negatives.

### Processor Revision Id
I provided a test to the homebrew community that prints out their system's PRId register. All tested systems, including one PAL, produced 0xB22. It's safe to assume, until and exception is found, that this is likely what all n64 consoles will provide.

### Unused
The COP0 has several "unused" registers which aren't completely unusable. Writes to any of these registers, won't actually store anything, but the written value is held in some kind of internal "latch", and can be read back at a later time. Only writes to any COP0 register appears to change the "latched" value. Other instructions that store data to regular GPR's do not affect it.

Also, all unused registers are effectively interchangeable. Writing some value to Unused7, will match the readback from Unused31, etc. Writes to any used register (whether read-only or not), will match the readback from any Unused register.

### Parity and Cache Error
According to VR4300 docs, these registers only exist for "compatibility reasons" but also supposedly aren't used by the CPU. They do still exist though, so they do _not_ behave like the Unused registers. The lower 8-bits of Parity are also write-able.

### CODE REVIEW

**Comment by bigbass1997** [comment]
Hold off on this PR for a bit. Turns out there is another possible value for the Processor Revision Id: 0xB10. More details soon.

**Comment by bigbass1997** [comment]
Ok we should be good to go now, unless any review changes are requested.

**Review by lemmy-64 — CHANGES_REQUESTED** [review]
Awesome work, thanks a lot for contributing!

As we discussed, I'd like to find a solution for RandomReadEarly so that it doesn't become a part of the default testset, until we have a way to classify. Otherwise this looks all excellent!

**Pull request merged**
